src/io/flutter/perf/FlutterWidgetPerf.java

```diff
@@ -101,8 +101,10 @@ private void setRequestInProgress(boolean value) {
   }
 
   private void onFrame(ActionEvent event) {
-    for (EditorPerfModel decorations : editorDecorations.values()) {
-      decorations.onFrame();
+    synchronized (editorDecorations) {
```

Comment:
this is fine as a workaround. The right fix though is probably to  find which of these methods is being called on a background thread. It looks like onFrame is probably the culprit. Could make sense to add asserts for all the methods that we are on the UI thread to avoid the issue and add  `AsyncUtils.invokeLater(() -> {`
calls for any ones that are unexpectedly actually sometimes called on a background thread.